src/main/java/org/springframework/data/couchbase/core/ReactiveReplaceByIdOperationSupport.java

```diff
@@ -93,6 +97,13 @@ private ReplaceOptions buildReplaceOptions(T object) {
 			} else if (durabilityLevel != DurabilityLevel.NONE) {
 				options.durability(durabilityLevel);
 			}
+			if (expiry != null && !expiry.isZero()) {
+				options.expiry(expiry);
+			} else if (domainType.isAnnotationPresent(Document.class)) {
+				Document documentAnn = domainType.getAnnotation(Document.class);
+				long durationSeconds = documentAnn.expiryUnit().toSeconds(documentAnn.expiry());
```

Comment:
could some of these values (like the expiry unit be null and cause a NPE if not set?)

Reply:
Document annotation has a default expiryUnit of SECONDS and a default expiry of 0